src/map.rs

```diff
@@ -90,6 +96,103 @@ impl<T> Into<Option<T>> for RevOption<T> {
     }
 }
 
+pub enum Range<'a, 'b, K: 'a, V: 'a> where 'a: 'b {
```

Comment:
Hmm, is there a reason the lifetimes have to be like this? Why not `K: 'b, V: 'b`?